### ISSUE TITLE
Conflict In Ada When Leader Mapped to Space

### ISSUE THREAD

**Issue by chadSpeaksBootDev** [neovim/neovim#28900]
### Problem

I started programming in Ada and noticed an odd behaviour where there is a delay after pressing space at the end of every Ada keyword like `with`, `use`, `function`, `begin`, etc.

After disabling my keymap in my lua configs, the issue goes away. The only thing in my keymap is a bind of leader to space.

Ada is the only language this happens in.

### Steps to reproduce

My keymaps.lua:

```lua
vim.g.mapleader = ' '
vim.keymap.set('n', "<leader>pv", vim.cmd.Ex)
```

### Expected behavior

I expect that Ada should behave like the other languages I use where this is not an issue. C, Haskell, OCaml, Swift, etc.

### Neovim version (nvim -v)

v0.10.0

### Vim (not Nvim) behaves the same?

unsure

### Operating system/version

macOS 14.5 and Arch Linux

### Terminal name/version

Alacritty 0.13.2

### $TERM environment variable

xterm-256colour

### Installation

Brew and Pacman

**Comment by zeertzjq** [comment]
It seems that this has been fixed in https://github.com/krischik/vim-ada/commit/32e5b9edc56856fa400575691ade4486215f7847 but the fix still hasn't been add to Vim runtime.

**Comment by fsbr** [comment]
How do I add the fix to the vim runtime? I just encountered this exact issue today and was wondering if there's a way to either log the behaviour of what is happening in neovim when I press the leader key in an Ada file.

**Comment by sully-vian** [comment]
Same problem here. Should we contact the maintainer of the `autoload/ada.vim` script ?

**Comment by clason** [comment]
Yes.

**Comment by krischik** [comment]
> It seems that this has been fixed in [krischik/vim-ada@32e5b9e](https://github.com/krischik/vim-ada/commit/32e5b9edc56856fa400575691ade4486215f7847) but the fix still hasn't been add to Vim runtime.

Yes, an update to the newest version might be helpful: https://github.com/krischik/vim-ada. However, I'm **not** a neovim runtime maintainer as I use GVim. I have absolutely no idea how to update the runtime. Last time I just send a ZIP to Bram and he did the rest.

I did make the bundle Deim, Plug, etc compatible so it can be imported via you favourite plugin manager which might be the easier option. Which is in fact what I do and makes more sense for a large bundle.

Also note that this is a bundle consisting of a dozen scripts. You can't just replace one part. You have to replace it all.

**Comment by clason** [comment]
>  I have absolutely no idea how to update the runtime. Last time I just send a ZIP to Bram and he did the rest.

This Is The Way, except now chrisbra has taken over as main maintainer and prefers to work via Github.

So the best thing is to open a PR to vim/vim updating the bundled Ada runtime files (ftplugin, autoload, syntax, indent, whatever you have) from your plugin (pointing out that you are the runtime file maintainer, in case it's not obvious from the file headers). There may be some requests for changes to match general runtime file standards, but usually such updates are merged quickly.

If you don't want to do that, explicitly give someone else willing to make such a PR permission (comment in an issue is enough, as long as it can be linked from the PR.)

Once Vim updates their runtime files, we will port the patches as a matter of course.
